Mathlib/Algebra/Category/ModuleCat/Subobject.lean

```diff
@@ -108,9 +106,7 @@ theorem cokernel_π_imageSubobject_ext {L M N : ModuleCat.{v} R} (f : L ⟶ M) [
   subst w
   -- Porting note (https://github.com/leanprover-community/mathlib4/pull/10959): The proof from here used to just be `simp`.
   simp only [map_add, add_right_eq_self]
-  change ((cokernel.π g) ∘ (g) ∘ (factorThruImageSubobject f)) l = 0
-  rw [← coe_comp, ← coe_comp, Category.assoc]
-  simp only [cokernel.condition, comp_zero]
-  rfl
+  change ((factorThruImageSubobject f) ≫ g ≫ (cokernel.π g)).hom l = 0
+  simp
```

Comment:
The goal after `simp` here is 
```
(cokernel.π g).hom (g.hom ((factorThruImageSubobject f).hom l)) = 0
```
I think we should just have a simp lemma for `(cokernel.π g).hom (g.hom x) = 0`, called something like `ModuleCat.Hom.cokernel_condition`.

Ideally we'd have some tactic help, e.g. like `@[elementwise]`, but for now we should just write these at the point we set up cokernels.

We might also think (by introducing a level of indirection to `ModuleCat.Hom.hom`?) about how we could write a single simp lemma that would apply `cokernel_condition` to all of its different concrete versions.

Reply:
I added a TODO comment as per our discussion.